### PR TITLE
[PWN-8202] Rewrite ChoosePhoneCode and DerivableAccounts view from UIKit to SwiftUI

### PR DESCRIPTION
**Pull request by lisemyon**
## Link jira to issue
https://p2pvalidator.atlassian.net/browse/PWN-8202

## Description of the changes
- I have changed only views of the screens. ViewModels have remained almost the same. I have not touched methods from old BECollectionViewModel as I think it should be done in another task. Apparently we have AsyncList, it might fit.
- I had to add [UITextField+Combine.swift](https://github.com/p2p-org/p2p-wallet-ios/compare/feature/pwn-7466...feature/PWN-8202?expand=1#diff-4a1429698905037c2605068b6ec08cd9eeff664a065a94bb3a438449c928c75a) because BECollectionView_Combine had a dependency of CocoaCombine. 'textPublisher' property is used in 2 places: EnterPhoneCode and EnterSMSCode screens, they probably should be rewritten two someday in SwiftUI and it is not worth to put CocoaCombine dependency because of these 2 screens.
- I tried to remove dependency directly but I have not found Package.swift for the project, to be honest

### CODE REVIEW

File: p2p_wallet/Models/DerivableAccount.swift [hunk 1]
```diff
@@ -8,12 +8,16 @@
...
+struct DerivableAccount: Hashable, Identifiable {
     let derivablePath: DerivablePath
     let info: KeyPair
     var amount: Double?
     var price: Double?
 
     // additional
     var isBlured: Bool?
+    
+    var id: String {
```
Comment (bigearsenal) [inline p2p_wallet/Models/DerivableAccount.swift:20]: just info.publicKey is enough, as each derivablePath has its own publicKey

**Review by bigearsenal — CHANGES_REQUESTED** [review]
Lets remove the BECollectionView, BECollectionView_Combine from project.yml also

**Comment by bigearsenal** [comment]
> ## Link jira to issue
> https://p2pvalidator.atlassian.net/browse/PWN-8202
> 
> ## Description of the changes
> * I have changed only views of the screens. ViewModels have remained almost the same. I have not touched methods from old BECollectionViewModel as I think it should be done in another task. Apparently we have AsyncList, it might fit.
> * I had to add [UITextField+Combine.swift](https://github.com/p2p-org/p2p-wallet-ios/compare/feature/pwn-7466...feature/PWN-8202?expand=1#diff-4a1429698905037c2605068b6ec08cd9eeff664a065a94bb3a438449c928c75a) because BECollectionView_Combine had a dependency of CocoaCombine. 'textPublisher' property is used in 2 places: EnterPhoneCode and EnterSMSCode screens, they probably should be rewritten two someday in SwiftUI and it is not worth to put CocoaCombine dependency because of these 2 screens.
> * I tried to remove dependency directly but I have not found Package.swift for the project, to be honest

Absolutely well done

File: p2p_wallet/Models/DerivableAccount.swift (hunk 1 above)
Comment (lisemyon) [reply p2p_wallet/Models/DerivableAccount.swift:20]: fixed

**Review by bigearsenal — CHANGES_REQUESTED** [review]
image
couldn't click the entire panel

**Comment by lisemyon** [comment]
> image
> 
> couldn't click the entire panel (padding, Spacer)

should be fixed now

**Pull request merged**